lib/sales_engine.rb

```diff
@@ -240,33 +244,34 @@ def merchant_items(date="all")
   end
 
   def find_most_revenue_from_merchant_repository(x)
-    merchant_ids = merchant_revenue.sort[-x..-1].collect {|i| i[1]}
+    merchant_ids = merchant_revenue.sort[-x..-1].collect { |i| i[1] }
     merchants = merchant_ids.map do |merchant_id|
-      merchant_repository.find_by_merchant_id(merchant_id)
+      merchant_repository.find_by_id(merchant_id)
     end
     merchants.reverse
   end
 
   def find_most_items_sold_from_merchant_repository(x)
-    merchant_ids = merchant_items.sort[-x..-1].collect {|i| i[1]}
+    merchant_ids = merchant_items.sort[-x..-1].collect { |i| i[1] }
     merchants = merchant_ids.map do |merchant_id|
-      merchant_repository.find_by_merchant_id(merchant_id)
+      merchant_repository.find_by_id(merchant_id)
     end
     merchants.reverse
   end
 
   def find_revenue_by_date_from_merchant_repository(date)
-    merchant_revenue(date).collect {|i| i[0]}.reduce(:+)
+    merchant_revenue(date).collect { |i| i[0] }.reduce(:+)
   end
 
-  def find_most_revenue_items(x)
-
+  def all_transactions
     all_invoices = invoice_repository.all
-      invoice_ids = all_invoices.map(&:id)
-    all_transactions = invoice_ids.map do |invoice_id|
+    invoice_ids = all_invoices.map(&:id)
+    invoice_ids.map do |invoice_id|
       transaction_repository.find_all_by_invoice_id(invoice_id)
     end
+  end
 
+  def find_most_revenue_items(x)
```

Comment:
Assignment Branch Condition size for find_most_revenue_items is too high. [37/15]